tests/test_qnode.py

```diff
@@ -530,64 +530,64 @@ def classnode(w, x=None):
-        self.logTestName()
+        multidim_array = np.reshape(b, s)
+        def circuit(w):
+            qml.RX(w[np.unravel_index(0,s)], wires=0) # b[0]
+            qml.RX(w[np.unravel_index(1,s)], wires=1) # b[1]
+            qml.RX(w[np.unravel_index(2,s)], wires=2) # ...
+            qml.RX(w[np.unravel_index(3,s)], wires=3)
+            qml.RX(w[np.unravel_index(4,s)], wires=4)
+            qml.RX(w[np.unravel_index(5,s)], wires=5)
+            qml.RX(w[np.unravel_index(6,s)], wires=6)
+            qml.RX(w[np.unravel_index(7,s)], wires=7)
+            return tuple(qml.expval.PauliZ(idx) for idx in range(len(b)))
 
-        for s in b_shapes:
-            multidim_array = np.reshape(b, s)
-            def circuit(w):
-                qml.RX(w[np.unravel_index(0,s)], wires=0) # b[0]
-                qml.RX(w[np.unravel_index(1,s)], wires=1) # b[1]
-                qml.RX(w[np.unravel_index(2,s)], wires=2) # ...
-                qml.RX(w[np.unravel_index(3,s)], wires=3)
-                qml.RX(w[np.unravel_index(4,s)], wires=4)
-                qml.RX(w[np.unravel_index(5,s)], wires=5)
-                qml.RX(w[np.unravel_index(6,s)], wires=6)
-                qml.RX(w[np.unravel_index(7,s)], wires=7)
-                return tuple(qml.expval.PauliZ(idx) for idx in range(len(b)))
-            circuit = qml.QNode(circuit, self.dev8)
-
-            # circuit evaluations
-            circuit_output = circuit(multidim_array)
-            expected_output = np.cos(b)
-            self.assertAllAlmostEqual(circuit_output, expected_output, delta=self.tol)
-
-            # circuit jacobians
-            circuit_jacobian = circuit.jacobian(multidim_array)
-            expected_jacobian = -np.diag(np.sin(b))
-            self.assertAllAlmostEqual(circuit_jacobian, expected_jacobian, delta=self.tol)
+        dev = qml.device('default.qubit', wires=8)
+        circuit = qml.QNode(circuit, dev)
 
+        # circuit evaluations
+        circuit_output = circuit(multidim_array)
+        expected_output = np.cos(b)
+        assert np.allclose(circuit_output, expected_output, atol=tol, rtol=0)
+
+        # circuit jacobians
+        circuit_jacobian = circuit.jacobian(multidim_array)
+        expected_jacobian = -np.diag(np.sin(b))
+        assert np.allclose(circuit_jacobian, expected_jacobian, atol=tol, rtol=0)
 
     def test_qnode_cv_gradient_methods(self):
-        "Tests the gradient computation methods on CV circuits."
-        # we can only use the 'A' method on parameters which only affect gaussian operations that are not succeeded by nongaussian operations
-        self.logTestName()
+        """Tests the gradient computation methods on CV circuits."""
+        # we can only use the 'A' method on parameters which only affect gaussian operations
+        # that are not succeeded by nongaussian operations
 
         par = [0.4, -2.3]
+        dev = qml.device('default.qubit', wires=2)
+
         def check_methods(qf, d):
```

Comment:
I would prefer to reduce the function-in-function since this is just two lines of code. It is probably a matter of choice but duplication seems to be a bit nicer to read than the function-in-function.

```
q = qml.QNode(qf, dev)
# NOTE: the default plugin is a discrete (qubit) simulator, it cannot
# execute CV gates, but the QNode can be constructed
q.construct(par)
assert q.grad_method_for_par == d
```

I am also thinking if we could take the `qf` function outside the test class since it is just an independent function with args (x, y).

Reply:
I agree with removing any function (such as `check_methods`) that contain `assert` statements for clarity.

Reply:
Let's move this into a new issue (or perhaps a comment in #99), to be completed once the second half of `test_qnode.py` is ported over.